addons/addon-base-raas/packages/base-raas-services/lib/study/__tests__/study-operation-service.test.js

```diff
@@ -0,0 +1,262 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const ServicesContainer = require('@aws-ee/base-services-container/lib/services-container');
+const { getSystemRequestContext } = require('@aws-ee/base-services/lib/helpers/system-context');
+
+// Mocked services
+jest.mock('@aws-ee/base-services/lib/db-service');
+jest.mock('@aws-ee/base-services/lib/lock/lock-service');
+jest.mock('@aws-ee/base-services/lib/logger/logger-service');
+jest.mock('@aws-ee/base-services/lib/settings/env-settings-service');
+jest.mock('@aws-ee/base-services/lib/plugin-registry/plugin-registry-service');
+jest.mock('@aws-ee/base-services/lib/audit/audit-writer-service');
+jest.mock('../../environment/service-catalog/environment-sc-service');
+jest.mock('../../user/user-service');
+jest.mock('../../study/study-service');
+jest.mock('../../study/study-permission-service');
+
+const Aws = require('@aws-ee/base-services/lib/aws/aws-service');
+const Logger = require('@aws-ee/base-services/lib/logger/logger-service');
+const DbService = require('@aws-ee/base-services/lib/db-service');
+const LockService = require('@aws-ee/base-services/lib/lock/lock-service');
+const PluginRegistryService = require('@aws-ee/base-services/lib/plugin-registry/plugin-registry-service');
+const SettingsService = require('@aws-ee/base-services/lib/settings/env-settings-service');
+const AuthService = require('@aws-ee/base-services/lib/authorization/authorization-service');
+const AuditService = require('@aws-ee/base-services/lib/audit/audit-writer-service');
+const JsonSchemaValidationService = require('@aws-ee/base-services/lib/json-schema-validation-service');
+const UserService = require('../../user/user-service');
+const EnvironmentScService = require('../../environment/service-catalog/environment-sc-service');
+const StudyService = require('../study-service');
+const StudyPermissionService = require('../study-permission-service');
+const StudyOperationService = require('../study-operation-service');
+
+const createResearcherContext = ({ uid = 'uid-researcher-1' } = {}) => ({
+  principalIdentifier: { uid },
+  principal: { userRole: 'researcher', status: 'active' },
+});
+
+describe('StudyOperationService', () => {
```

Comment:
Can you add a test for the case when updating an environment fails but we go through other environments successfully.